runtime/constants.go

```diff
@@ -34,6 +34,9 @@ const (
 	// MetricEndpointPanics is endpoint level panic counter
 	MetricEndpointPanics = "endpoint.panic"
 
+	// MetricEndpointAppErrors is endpoint level panic counter
+	MetricEndpointAppErrors = "endpoint.app-errors-err"
```

Comment:
Maybe just `endpoint.app-errors`?

Reply:
`endpoint.app-errors` already exists. There is some generic code outside the template that processes the TChannel server result. Because it's outside the template it isn't straightforward to add a tag per error type. It would be possible to just remove the other stat and just emit the stat once with the caveat that it makes the generic code a bit more complex by omission.

Reply:
+1